android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java

```diff
@@ -53,64 +53,135 @@ of this software and associated documentation files (the "Software"), to deal
+            }
+        }
+
+        @Override
+        public void onDidDisplay(IInAppMessageDidDisplayEvent event) {
+            try {
+                sendEvent("OneSignal-inAppMessageDidDisplay", RNUtils.convertHashMapToWritableMap(RNUtils.convertInAppMessageToMap(event.getMessage())));
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+
+        @Override
+        public void onWillDismiss(IInAppMessageWillDismissEvent event) {
+            try {
+                sendEvent("OneSignal-inAppMessageWillDismiss", RNUtils.convertHashMapToWritableMap(RNUtils.convertInAppMessageToMap(event.getMessage())));
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+
+        @Override
+        public void onDidDismiss(IInAppMessageDidDismissEvent event) {
+            try {
+                sendEvent("OneSignal-inAppMessageDidDismiss", RNUtils.convertHashMapToWritableMap(RNUtils.convertInAppMessageToMap(event.getMessage())));
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+    };
+
+    private INotificationClickListener rnNotificationClickListener = new INotificationClickListener() {
+        @Override
+        public void onClick(INotificationClickEvent event) {
+            try {
+                sendEvent("OneSignal-notificationClicked", RNUtils.convertHashMapToWritableMap(RNUtils.convertNotificationToMap(event.getNotification())));
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+    };
 
     private void removeObservers() {
-        this.removePermissionChangedHandler();
-        this.rmeovePushSubscriptionChangeHandler();
+        this.removePermissionObserver();
+        this.removePushSubscriptionObserver();
     }
 
     private void removeHandlers() {
-        OneSignal.getInAppMessages().setInAppMessageClickHandler(null);
-        OneSignal.getInAppMessages().setInAppMessageLifecycleHandler(null);
-        OneSignal.getNotifications().setNotificationClickHandler(null);
-        OneSignal.getNotifications().setNotificationWillShowInForegroundHandler(null);
+        OneSignal.getInAppMessages().removeClickListener(rnInAppClickListener);
+        OneSignal.getInAppMessages().removeLifecycleListener(rnInAppLifecycleListener);
+        OneSignal.getNotifications().removeClickListener(rnNotificationClickListener);
+        OneSignal.getNotifications().removeForegroundLifecycleListener(this);
     }
 
     private void sendEvent(String eventName, Object params) {
         mReactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit(eventName, params);
     }
 
-    private void initNotificationWillShowInForegroundHandlerParams() {
-        this.hasSetNotificationWillShowInForegroundHandler = true;
+    private void initNotificationWillDisplayInForegroundListenerParams() {
```

Comment:
I think method no longer used, can remove

Reply:
Addressed in [Update Android Notification and IAM listeners to send entire event](https://github.com/OneSignal/react-native-onesignal/pull/1532/commits/762ee2b05d3c4c6e272c7c4a3b994f0e00575b59)